lightning-invoice/src/payment.rs

```diff
@@ -487,7 +455,7 @@ where
 		let inflight_htlcs = self.create_inflight_map();
 		let route = self.router.find_route(
 			&payer, &params, &payment_hash, Some(&first_hops.iter().collect::<Vec<_>>()),
-			&AccountForInFlightHtlcs { scorer: &mut self.scorer.lock(), inflight_htlcs }
+			InFlightHtlcs::new(Some(inflight_htlcs))
```

Comment:
Let's have `create_inflight_map` return this instead of wrapping at each call site. Also, rename to `create_inflight_htlcs`

Reply:
Addressed in 441f28ee177c101bd5ee68ff3273472e37f14690